app/Models/Thread.php

```diff
@@ -150,4 +150,19 @@ public static function feedQuery(): Builder
                 END AS latest_creation
             '));
     }
+
+    /**
+     * This will calculate the average resolution time in days of all threads marked as resolved.
+     */
+    public static function resolutionTime()
+    {
+        try {
+            return static::join('replies', 'threads.solution_reply_id', '=', 'replies.id')
+                ->select(DB::raw('avg(datediff(replies.created_at, threads.created_at)) as duration'))
+                ->pluck('duration')
+                ->first();
+        } catch (\Exception $e) {
```

Comment:
Import this with a use statement.

Reply:
Why does this throw an exception in the first place?

Reply:
The datediff function is not available in SQLite so tests fail. Happy to approach this in a different way if you have any ideas?

Reply:
Since they're timestamps, you should be able to just do a subtraction which should work in sqlite as well:

```sql
avg(replies.created_at - threads.created_at) as duration
```

Reply:
You can't do that in MySQL

Reply:
I just tried it out and it worked. What version are you using?

Reply:
I'm using MariaDB 10.3.14 which I think is equivalent to MySQL 5.7.

What result did it give you?

Reply:
I just tried it out on a different DB with two timestamps, not on the production DB. We're not using MariaDB for Laravel.io so can you test on MySQL? Using 5.7 for Laravel.io

Reply:
I just ran this on my blog which is running MySQL 5.7:

```
SELECT
	posts.created_at - users.created_at as diff1,
	DATEDIFF(posts.created_at, users.created_at) as diff2
FROM
	users
	JOIN posts ON posts.user_id = users.id;
```

and got the following results:

| diff1 | diff2 |
| --- | --- |
| 9098993794 | 91 |
| 9099944312 | 92 |
| 9099944639 | 92 |
| 9099944903 | 92 |
| 9099945185 | 92 |

Reply:
I ran the following on the production DB:

```sql
SELECT avg(replies.created_at - threads.created_at) as duration
FROM threads
JOIN replies ON threads.solution_reply_id = replies.id
```

Which gives me: `83805418.8335`. Which is...

83805418.8335 / 60 / 60 / 24 = 969.970125388 days 😅 

Maybe not a good statistic to have after all 😂 

But I think the above will work in sqlite as well?

Reply:
I’m pretty sure that query gives the wrong result in MySQL though?

Reply:
A lot of threads have very late replies so I'm not sure if it's wrong. I do see quite a few other suggestions for determining time difference on stackoverflow. Does any of the following also work on sqlite? https://stackoverflow.com/questions/4759248/difference-between-two-dates-in-mysql